tests/filter.json

```diff
@@ -813,6 +821,14 @@
         {"a": 100, "d": "e"}
       ]
     },
+    {
+      "name": "equals number, exponent upper e",
+      "selector" : "$[?@.a==1E2]",
```

Comment:
This isn't valid according to the ABNF

```abnf
number              = (int / "-0") [ frac ] [ exp ] ; decimal number
frac                = "." 1*DIGIT                  ; decimal fraction
exp                 = "e" [ "-" / "+" ] 1*DIGIT    ; decimal exponent
true                = %x74.72.75.65                ; true
false               = %x66.61.6c.73.65             ; false
null                = %x6e.75.6c.6c                ; null
```

Upper `E` is not listed as an allowed exponent character.

Reply:
`E` seems to be valid (same also for JSON), see https://www.rfc-editor.org/rfc/rfc9535#section-2.3.5.1-13:
> Note: Alphabetic characters in quoted strings are case-insensitive in ABNF, so within a floating point number, the ABNF expression "e" can be either the character 'e' or 'E'.